leadbutt.py

```diff
@@ -173,50 +351,90 @@ def get_metric_statistics(**kwargs):
         connect_args['aws_secret_access_key'] = auth_options['aws_secret_access_key']
     conn = boto.ec2.cloudwatch.connect_to_region(region, **connect_args)
-    for metric in config['Metrics']:
-        options = get_options(
-            config_options, metric.get('Options'), cli_options)
+    if 'Metrics' in config:
+        for metric in config['Metrics']:
+            options = get_options(
+                config_options, metric.get('Options'), cli_options)
+            period_local = options['Period'] * 60
+            count_local = options['Count']
+            # if you have metrics that are available only every 5 minutes, be sure to request only stats
+            # that are likely/sure to be up to date, ie ones ending on the previous
+            # period increment.
+            end_time = datetime.datetime.utcnow() - datetime.timedelta(seconds=int(time.time()) % period_local)
+            start_time = end_time - datetime.timedelta(seconds=period_local * count_local)
+
+            # if 'Unit 'is in the config, request only that; else get all units
+            unit = metric.get('Unit')
+            metric_names = metric['MetricName']
+            if not isinstance(metric_names, list):
+                metric_names = [metric_names]
+            for metric_name in metric_names:
+                # we need a copy of the metric dict with the MetricName swapped out
+                this_metric = metric.copy()
+                this_metric['MetricName'] = metric_name
+                results = get_metric_statistics(
+                    connection=conn,
+                    period=period_local,
+                    start_time=start_time,
+                    end_time=end_time,
+                    metric_name=metric_name,
+                    namespace=metric['Namespace'],
+                    statistics=metric['Statistics'],
+                    dimensions=metric['Dimensions'],
+                    unit=unit
+                )
+
+                if 'NullIsZero' in options and metric_name in options['NullIsZero']:
+                    results = value_pad_results(
+                        results,
+                        start_time,
+                        end_time,
+                        options['NullIsZero'][metric_name],
+                    )
+
+                output_results(results, this_metric, options)
+                time.sleep(kwargs.get('interval', 0) / 1000.0)
+
+    # get enhanced monitoring if it is enabled
+    if enhanced_monitoring:
+        options = get_options(config_options, None, cli_options)
+        # convert minutes to seconds
         period_local = options['Period'] * 60
         count_local = options['Count']
+        log_group = enhanced_monitoring['LogGroup']
+        # determine formatter
+        if 'Formatter' in enhanced_monitoring:
+            options['Formatter'] = enhanced_monitoring['Formatter']
+            options['ListFormatter'] = options['Formatter']
```

Comment:
I made a mistake here on line 408. It should be options['ListFormatter'] = enhanced_monitoring['Formatter'].

I want to make the config file as easy to setup as before, but if the user wants granularity they can setup a custom formatter for each type of log. I want to overwrite the formatter in the enhanced monitoring section if they were provided in the yaml file.